app/processing/scheduled/get-pending-payment-requests.js

```diff
@@ -1,17 +1,45 @@
 const moment = require('moment')
 const db = require('../../data')
 
-const getPendingPaymentRequests = async (started, transaction) => {
-  return db.schedule.findAll({
-    where: {
-      completed: null,
-      started: { [db.Sequelize.Op.gt]: moment(started).subtract(5, 'minutes').toDate() }
-    },
-    include: [{
-      model: db.paymentRequest,
-      as: 'paymentRequest'
-    }],
-    transaction
+const getPendingPaymentRequests = async (scheduledPaymentRequests, started, transaction) => {
+  if (scheduledPaymentRequests.length === 0) {
+    return []
+  }
+
+  const scheduledPropsToCheck = scheduledPaymentRequests.map(spr => ({
+    frn: spr.paymentRequest.frn,
+    schemeId: spr.paymentRequest.schemeId,
+    marketingYear: spr.paymentRequest.marketingYear
+  }))
+
+  const replacements = { startedMoment: moment(started).subtract(5, 'minutes').toDate() }
+  const conditions = []
+  scheduledPropsToCheck.forEach((sch, idx) => {
+    conditions.push(`("paymentRequests"."frn" = :frn${idx} AND "paymentRequests"."schemeId" = :schemeId${idx} AND "paymentRequests"."marketingYear" = :marketingYear${idx})`)
+    replacements[`frn${idx}`] = sch.frn
+    replacements[`schemeId${idx}`] = sch.schemeId
+    replacements[`marketingYear${idx}`] = sch.marketingYear
+  })
```

Comment:
Could we use `.map` to simplify this?

Reply:
Similar sort of code but have changed to map which agree is narrowly cleaner